airflow-core/src/airflow/ui/src/components/Graph/DownloadButton.tsx

```diff
@@ -26,24 +26,28 @@ import { toaster } from "src/components/ui";
 
 export const DownloadButton = ({ name }: { readonly name: string }) => {
   const { t: translate } = useTranslation("components");
-  const { getNodes, getZoom } = useReactFlow();
 
-  const onClick = () => {
-    const nodesBounds = getNodesBounds(getNodes());
+  const { fitView } = useReactFlow();
+
+  const onClick = async () => {
+
+    // Ensure the graph fits before taking screenshot
+    fitView({ duration: 0, padding: 0.1 });
+
+    // Small delay to let layout stabilize
+    await new Promise((resolve) => setTimeout(resolve, 50));
```

Comment:
This is really not great. I don't think we need that. `fitView` should return a promise, and we can await that to be sure that we move forward once the graph fits.